### ISSUE TITLE
Host or domain name not found. Name service error for name=outlook-com.olc.protection.outlook.com type=AAAA: Host found but no data record of requested type

### ISSUE THREAD

**Issue by Aaron-Ritter** [Mailu/Mailu#2934]
Host or domain name not found. Name service error for name=outlook-com.olc.protection.outlook.com type=AAAA: Host found but no data record of requested type.

## Environment & Version

Mailu 2.0.23

### Environment

- [x] docker compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: ghcr.io/mailu/*:2.0

## Description
For some reason it is trying to lookup AAAA record when sending to @outlook.com emails which does not exist regularly. And i get a client side delivery error because of it: `Host or domain name not found. Name service error for name=outlook-com.olc.protection.outlook.com type=AAAA: Host found but no data record of requested type`

```
mailu-smtp-1       | Sep 05 21:49:08 mx01 postfix/smtpd[9610]: 13031C00E7: client=gatekeeper02.masked.toplvl[x.x.x.205], sasl_method=XCLIENT, sasl_username=website@customer-domain.toplvl
mailu-smtp-1       | Sep 05 21:49:08 mx01 postfix/cleanup[9616]: 13031C00E7: message-id=<message-id-masked@customer-domain.toplvl>
mailu-smtp-1       | Sep 05 21:49:08 mx01 postfix/qmgr[359]: 13031C00E7: from=<www-data@customer-domain.toplvl>, size=6225, nrcpt=1 (queue active)
mailu-smtp-1       | Sep 05 21:49:09 mx01 postfix/smtp[9622]: 13031C00E7: to=<masked-email-address@outlook.com>, relay=none, delay=1.2, delays=0.71/0.02/0.48/0, dsn=5.4.4, status=bounced (Host or domain name not found. Name service error for name=outlook-com.olc.protection.outlook.com type=AAAA: Host found but no data record of requested type)
mailu-smtp-1       | Sep 05 21:49:09 mx01 postfix/bounce[9623]: 13031C00E7: sender non-delivery notification: 2880AC00ED
mailu-smtp-1       | Sep 05 21:49:09 mx01 postfix/qmgr[359]: 13031C00E7: removed
mailu-antispam-1   | 2023-09-05 21:49:08 #544(rspamd_proxy) <d99915>; proxy; rspamd_message_parse: loaded message; id: <message-id-masked@customer-domain.toplvl>; queue-id: <13031C00E7>; size: 6179; checksum: <2acf899f17417671e60f3c04e59f95e0>
mailu-antispam-1   | 2023-09-05 21:49:08 #544(rspamd_proxy) <d99915>; proxy; rspamd_task_write_log: id: <message-id-masked@customer-domain.toplvl>, qid: <13031C00E7>, ip: x.x.x.205, user: website@customer-domain.toplvl, from: <www-data@customer-domain.toplvl>, (default: F (no action): [1.49/15.00] [ENVFROM_SERVICE_ACCT(1.00){},FORGED_SENDER(0.30){shop@customer-domain.toplvl;www-data@customer-domain.toplvl;},MIME_HTML_ONLY(0.20){},ARC_NA(0.00){},ASN(0.00){asn:24940, ipnet:159.69.0.0/16, country:DE;},DKIM_SIGNED(0.00){customer-domain.toplvl:s=dkim;},FREEMAIL_ENVRCPT(0.00){outlook.com;},FREEMAIL_TO(0.00){outlook.com;},FROM_HAS_DN(0.00){},FROM_NEQ_ENVFROM(0.00){shop@customer-domain.toplvl;www-data@customer-domain.toplvl;},HAS_REPLYTO(0.00){shop@customer-domain.toplvl;},IS_LOCAL_DOMAIN_E(0.00){customer-domain.toplvl;},IS_LOCAL_DOMAIN_H(0.00){customer-domain.toplvl;},MID_RHS_MATCH_FROM(0.00){},MIME_TRACE(0.00){0:~;},NEURAL_HAM(0.00){-0.990;},RCPT_COUNT_ONE(0.00){1;},RCVD_COUNT_ZERO(0.00){0;},REPLYTO_EQ_FROM(0.00){},TO_DN_NONE(0.00){},TO_MATCH_ENVRCPT_ALL(0.00){},BAYES_HAM(-0.00){27.71%;}]), len: 6179, time: 491.117ms, dns req: 16, digest: <2acf899f17417671e60f3c04e59f95e0>, rcpts: <masked-email-address@outlook.com>, mime_rcpts: <masked-email-address@outlook.com>
```

**Comment by nextgens** [comment]
This is something you should see with their postmaster.

```
$ host -t MX outlook.com.
outlook.com mail is handled by 5 outlook-com.olc.protection.outlook.com. 

$ host -t AAAA outlook-com.olc.protection.outlook.com.
outlook-com.olc.protection.outlook.com has no AAAA record

$ host -t A outlook-com.olc.protection.outlook.com.|wc -l
55
```

Odds are it's the 55 A records that are creating trouble and get lost "somewhere".

**Comment by nextgens** [comment]
If you do get somewhere with them you may want to express that serving 55 IPs, all in the same ASN, is stupid.

**Comment by Aaron-Ritter** [comment]
But why is it not doing the ipv4 lookup?

I get that their setup is crazy, but why does it stop trying after trying ipv6 (AAAA) shouldn't it at least try to get ipv4?

**Comment by nextgens** [comment]
I've just re-run the queries from above, now I get 2 A records instead of 55... same ASN but a different subnet.

I don't think we can investigate this further without involvement from their side.

**Comment by Aaron-Ritter** [comment]
I see your point, and especially with these inconsistent results, I'll try to debug it further.

> docker compose exec smtp host outlook.com

should host be in the default container build for smtp? I don't have it available.

**Comment by nextgens** [comment]
Ah yes, sorry I did it on the host:

```
# host -t A outlook-com.olc.protection.outlook.com. 1.1.1.1
Using domain server:
Name: 1.1.1.1
Address: 1.1.1.1#53
Aliases: 

outlook-com.olc.protection.outlook.com has address 52.101.40.30
outlook-com.olc.protection.outlook.com has address 52.101.11.5
outlook-com.olc.protection.outlook.com has address 52.101.42.7
outlook-com.olc.protection.outlook.com has address 52.101.9.1
outlook-com.olc.protection.outlook.com has address 52.101.73.29
outlook-com.olc.protection.outlook.com has address 52.101.11.8
outlook-com.olc.protection.outlook.com has address 52.101.11.6
outlook-com.olc.protection.outlook.com has address 52.101.73.20
outlook-com.olc.protection.outlook.com has address 52.101.73.25
outlook-com.olc.protection.outlook.com has address 52.101.73.17
outlook-com.olc.protection.outlook.com has address 52.101.9.3
outlook-com.olc.protection.outlook.com has address 52.101.9.13
outlook-com.olc.protection.outlook.com has address 52.101.137.1
outlook-com.olc.protection.outlook.com has address 52.101.137.3
outlook-com.olc.protection.outlook.com has address 52.101.8.33
outlook-com.olc.protection.outlook.com has address 52.101.73.14
outlook-com.olc.protection.outlook.com has address 52.101.68.37
outlook-com.olc.protection.outlook.com has address 52.101.40.25
outlook-com.olc.protection.outlook.com has address 52.101.40.3
outlook-com.olc.protection.outlook.com has address 52.101.73.7
outlook-com.olc.protection.outlook.com has address 52.101.40.28
outlook-com.olc.protection.outlook.com has address 52.101.11.11
outlook-com.olc.protection.outlook.com has address 52.101.42.11
outlook-com.olc.protection.outlook.com has address 52.101.132.31
outlook-com.olc.protection.outlook.com has address 52.101.73.23
outlook-com.olc.protection.outlook.com has address 52.101.40.7
outlook-com.olc.protection.outlook.com has address 52.101.132.29
outlook-com.olc.protection.outlook.com has address 52.101.73.13
outlook-com.olc.protection.outlook.com has address 52.101.40.5
outlook-com.olc.protection.outlook.com has address 52.101.73.5
outlook-com.olc.protection.outlook.com has address 52.101.42.5
outlook-com.olc.protection.outlook.com has address 52.101.68.1
outlook-com.olc.protection.outlook.com has address 52.101.9.10
outlook-com.olc.protection.outlook.com has address 52.101.68.19
outlook-com.olc.protection.outlook.com has address 52.101.68.2
outlook-com.olc.protection.outlook.com has address 52.101.68.20
outlook-com.olc.protection.outlook.com has address 52.101.73.31
outlook-com.olc.protection.outlook.com has address 52.101.73.27
outlook-com.olc.protection.outlook.com has address 52.101.8.35
outlook-com.olc.protection.outlook.com has address 52.101.73.0
outlook-com.olc.protection.outlook.com has address 52.101.68.33
outlook-com.olc.protection.outlook.com has address 52.101.68.4
outlook-com.olc.protection.outlook.com has address 52.101.73.3
outlook-com.olc.protection.outlook.com has address 52.101.73.9
outlook-com.olc.protection.outlook.com has address 52.101.11.4
outlook-com.olc.protection.outlook.com has address 52.101.42.12
outlook-com.olc.protection.outlook.com has address 52.101.42.8
outlook-com.olc.protection.outlook.com has address 52.101.8.37
outlook-com.olc.protection.outlook.com has address 52.101.8.43
outlook-com.olc.protection.outlook.com has address 52.101.73.18
outlook-com.olc.protection.outlook.com has address 52.101.8.45
outlook-com.olc.protection.outlook.com has address 52.101.8.47
outlook-com.olc.protection.outlook.com has address 52.101.9.15
outlook-com.olc.protection.outlook.com has address 52.101.73.10
outlook-com.olc.protection.outlook.com has address 52.101.68.17
```

**Comment by Aaron-Ritter** [comment]
in the meantime i have the issue with hotmail too, is ther any rule of retry or something like that i could adjusts? its obvious that it does not have IPv6 records, but id like it to retry the A record a few more times:

```
 [<address@hotmail.com>](mailto:address@hotmail.com): Host or domain name not found. Name service error
    for name=hotmail-com.olc.protection.outlook.com type=AAAA: Host found but
    no data record of requested type
```

**Comment by asaurer** [comment]
I'm having the same issues here with outlook.com. However, hotmail.com works for me:

```
postfix/smtp[813]: warning: no MX host for outlook.com has a valid address record
postfix/smtp[813]: 64EA7270417: to=<email@outlook.com>, relay=none, delay=1.5, delays=0.8/0.09/0.64/0, dsn=5.4.4, status=bounced (Host or domain name not found. Name service error for name=outlook-com.olc.protection.outlook.com type=A: Host found but no data record of requested type)
postfix/cleanup[816]: CC3012705E8: message-id=<20230924071928.CC3012705E8@server.mydomain.com>
postfix/bounce[815]: 64EA7270417: sender non-delivery notification: CC3012705E8
```

```
postfix/smtp[921]: Verified TLS connection established to hotmail-com.olc.protection.outlook.com[104.47.55.33]:25: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
postfix/smtp[921]: DA4042704CD: to=<email@hotmail.com>, relay=hotmail-com.olc.protection.outlook.com[104.47.55.33]:25, delay=5.4, delays=0.85/0.05/2.1/2.4, dsn=2.6.0, status=sent (250 2.6.0 <a05d1e162b0769b3000f25e8ce9ecadf67a6b7e3@mydomain.com> [InternalId=157736968921770, Hostname=VI1PR09MB4383.eurprd09.prod.outlook.com] 10020 bytes in 0.270, 36.199 KB/sec Queued mail for delivery -> 250 2.1.5)
postfix/qmgr[381]: DA4042704CD: removed
```

I remember resolving the issue with hotmail.com by contacting MS support and having my IP removed from a greylist. Not sure why it's working for hotmail.com and not for outlook.com. However, I can't find the email exchange I had with them. I have my IP registered with [Outlook.com Smart Network Data Services](https://sendersupport.olc.protection.outlook.com/snds/index.aspx)

I think I filled out this form https://olcsupport.office.com/ trying this now with outlook.com

**Comment by nextgens** [comment]
I have now run into it too... I am still not convinced that there is anything wrong on Mailu's side.

**Comment by Aaron-Ritter** [comment]
> I remember resolving the issue with hotmail.com by contacting MS support and having my IP removed from a greylist. Not sure why it's working for hotmail.com and not for outlook.com. However, I can't find the email exchange I had with them. I have my IP registered with [Outlook.com Smart Network Data Services](https://sendersupport.olc.protection.outlook.com/snds/index.aspx)

I registered the IPs and got "All of the specified IPs have normal status."

> 
> I think I filled out this form https://olcsupport.office.com/ trying this now with outlook.com

I tried it for hotmail for now, lets see.

**Comment by nextgens** [comment]
Keep DNS logs/packets  when this happens if you can.

``tcpdump -ni any -s0 -w /tmp/DNS.pcap port 53``

**Comment by nextgens** [comment]
@Aaron-Ritter https://www.mail-archive.com/postfix-users@postfix.org/msg89297.html

**Comment by nextgens** [comment]
@Aaron-Ritter I've now managed to reproduce it here.

Still looking but so far, this looks promising (broken SOA on NODATA replies):
https://dnsviz.net/d/hotmail-com.olc.protection.outlook.com/dnssec/

**Comment by nextgens** [comment]
Here is what it looks like when it works:

```
Sep 28 08:58:10 ocloud postfix/smtp[214836]: smtp_parse_destination: hotmail.com smtp
Sep 28 08:58:10 ocloud postfix/smtp[214836]: connecting to hotmail.com port 25
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_query: hotmail.com (MX): OK
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type MX for hotmail.com
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_query: . (NS): OK
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type NS for 
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dnssec_probe 'ns:.' received a response that is DNSSEC validated
Sep 28 08:58:10 ocloud postfix/smtp[214836]: smtp_addr_one: host hotmail-com.olc.protection.outlook.com
Sep 28 08:58:10 ocloud postfix/smtp[214836]: lookup hotmail-com.olc.protection.outlook.com type A flags RES_USE_DNSSEC
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_query: hotmail-com.olc.protection.outlook.com (A): OK
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type A for hotmail-com.olc.protection.outlook.com
Sep 28 08:58:10 ocloud postfix/smtp[214836]: dns_get_answer: type A for hotmail-com.olc.protection.outlook.com
Sep 28 08:58:10 ocloud postfix/smtp[214836]: lookup hotmail-com.olc.protection.outlook.com type AAAA flags RES_USE_DNSSEC
Sep 28 08:58:11 ocloud postfix/smtp[214836]: dns_query: hotmail-com.olc.protection.outlook.com (AAAA): Host found but no data record of requested type
Sep 28 08:58:11 ocloud postfix/smtp[214836]: begin hotmail.com address list
Sep 28 08:58:11 ocloud postfix/smtp[214836]: pref    2 host hotmail-com.olc.protection.outlook.com/104.47.18.161
Sep 28 08:58:11 ocloud postfix/smtp[214836]: pref    2 host hotmail-com.olc.protection.outlook.com/104.47.18.225
Sep 28 08:58:11 ocloud postfix/smtp[214836]: end hotmail.com address list
Sep 28 08:58:11 ocloud postfix/smtp[214836]: smtp_find_self: not found
Sep 28 08:58:11 ocloud postfix/smtp[214836]: global TLS level: dane
Sep 28 08:58:11 ocloud postfix/smtp[214836]: dict_sockmap_lookup: key hotmail.com
Sep 28 08:58:11 ocloud postfix/smtp[214836]: auto_clnt_open: connected to /tmp/podop.socket
Sep 28 08:58:11 ocloud postfix/smtp[214836]: dict_sockmap_lookup: key hotmail.com
Sep 28 08:58:11 ocloud postfix/smtp[214836]: auto_clnt_open: connected to /tmp/mta-sts.socket
Sep 28 08:58:11 ocloud postfix/smtp[214836]: maps_find: smtp_tls_policy_maps: socketmap:unix:/tmp/mta-sts.socket:postfix(0,lock|fold_fix|utf8_request): hotmail.com = secure match=.olc.protection.outlook.com servername=hostname
Sep 28 08:58:11 ocloud postfix/smtp[214836]: effective TLS level: secure
Sep 28 08:58:11 ocloud postfix/smtp[214836]: ctable_locate: install entry key hotmail.com:hotmail-com.olc.protection.outlook.com:25:
Sep 28 08:58:11 ocloud postfix/smtp[214836]: smtp_connect_addr: trying: hotmail-com.olc.protection.outlook.com[104.47.18.225] port 25...
Sep 28 08:58:11 ocloud postfix/smtp[214836]: vstream_tweak_tcp: TCP_MAXSEG 1448
Sep 28 08:58:11 ocloud postfix/smtp[214836]: fd=19: stream buffer size old=0 new=5792
Sep 28 08:58:11 ocloud postfix/smtp[214836]: smtp_stream_setup: maxtime=300 enable_deadline=0 min_data_rate=0
Sep 28 08:58:11 ocloud postfix/smtp[214836]: < hotmail-com.olc.protection.outlook.com[104.47.18.225]:25: 220 VI1EUR06FT043.mail.protection.outlook.com Microsoft ESMTP MAIL Service ready at Thu, 28 Sep 2023
```

... I'm still trying to reproduce it with the instrumentation turned on.

**Comment by hash070** [comment]
I encountered the same issue with Mailu 2.0.
Then I tried Mailu 1.9 and I found that version 1.9 doesn't have this problem.

**Comment by nextgens** [comment]
That's not surprising. A lot has changed DNS-resolution wise to enable DANE support.

**Comment by nextgens** [comment]
I have now managed to reproduce the problem with instrumentation.

```
Sep 28 16:24:44 email01 postfix/smtp[221104]: smtp_parse_destination: hotmail.com smtp
Sep 28 16:24:44 email01 postfix/smtp[221104]: connecting to hotmail.com port 25
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_query: hotmail.com (MX): OK
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type MX for hotmail.com
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_query: . (NS): OK
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_get_answer: type NS for 
Sep 28 16:24:44 email01 postfix/smtp[221104]: dnssec_probe 'ns:.' received a response that is DNSSEC validated
Sep 28 16:24:44 email01 postfix/smtp[221104]: smtp_addr_one: host hotmail-com.olc.protection.outlook.com
Sep 28 16:24:44 email01 postfix/smtp[221104]: lookup hotmail-com.olc.protection.outlook.com type A flags RES_USE_DNSSEC
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_query: hotmail-com.olc.protection.outlook.com (A): Host found but no data record of requested type
Sep 28 16:24:44 email01 postfix/smtp[221104]: lookup hotmail-com.olc.protection.outlook.com type AAAA flags RES_USE_DNSSEC
Sep 28 16:24:44 email01 postfix/smtp[221104]: dns_query: hotmail-com.olc.protection.outlook.com (AAAA): Host found but no data record of requested type
Sep 28 16:24:44 email01 postfix/smtp[221104]: warning: no MX host for hotmail.com has a valid address record
Sep 28 16:24:44 email01 postfix/smtp[221104]: connect to subsystem private/bounce
```

... the my pcap capture tells me that unbound has made the right query, got an answer and has passed it to the stub.

The problem is in between unbound and postfix, so this is definitely a bug; probably not our bug but a bug nevertheless.

@Aaron-Ritter thank you for insisting :)

**Comment by Aaron-Ritter** [comment]
thanks so much for the update and digging into it @nextgens !

how close to a failure does the command need to run?

**Comment by nextgens** [comment]
The only limit is the disk space available :)

Here I keep logs for 24h as I am likely to react fast if I see a bounce message with that error.

Ideally we would get three pieces:
- the bounce message (that has a timestamp we need)
- the DNS packets exchanged in between unbound and the SMTP container around the timestamp 
- the debug logs from dockerd related to around the timestamp

**Comment by nextgens** [comment]
So we got to the bottom of it. This is a musl bug... that's fixed in newer alpine images. When a truncated DNS reply is sent by unbound, it does not retry with TCP like it ought to.

https://github.com/alpinelinux/docker-alpine/issues/326 
We can't fix it with unbound: https://github.com/NLnetLabs/unbound/issues/621 
The patch is at  https://git.musl-libc.org/cgit/musl/commit/?id=51d4669fb97782f6a66606da852b5afd49a08001 but even newer versions of 3.17 do not have it.

```
root@a2:/tmp/o/testbed# docker run --rm -it --network=testbed_default --dns 192.168.204.254 alpine:3.17.2 /bin/sh
/ # ping 50x.example.com
ping: bad address '50x.example.com'
/ # ping 30x.example.com
ping: bad address '30x.example.com'
/ # ping 5x.example.com
PING 5x.example.com (127.0.0.101): 56 data bytes
64 bytes from 127.0.0.101: seq=0 ttl=64 time=0.029 ms
^C
--- 5x.example.com ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 0.029/0.029/0.029 ms
/ # 
root@a2:/tmp/o/testbed# docker run --rm -it --network=testbed_default --dns 192.168.204.254 alpine:3.18.4 /bin/sh
/ # ping 50x.example.com
PING 50x.example.com (127.0.0.133): 56 data bytes
64 bytes from 127.0.0.133: seq=0 ttl=64 time=0.030 ms
```

**Comment by Aaron-Ritter** [comment]
Uff, what a great find! and that's complex. Reading through the different issues it is mentioned `I found the solution, it is to install bind-tools package.`

Is this maybe something we could try in the unbound container as a temporary solution?

**Comment by nextgens** [comment]
Unfortunately that won't solve anything.

```
root@a2:/tmp/o/testbed# docker run --rm -it --network=testbed_default --dns 192.168.204.254 alpine:3.17 /bin/sh
/ # ping 5x.example.com
PING 5x.example.com (127.0.0.100): 56 data bytes
64 bytes from 127.0.0.100: seq=0 ttl=64 time=0.030 ms
64 bytes from 127.0.0.100: seq=1 ttl=64 time=0.062 ms
^C
--- 5x.example.com ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 0.030/0.046/0.062 ms
/ # ping 30x.example.com
ping: bad address '30x.example.com'
/ # apk add bind-tools
fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/aarch64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/aarch64/APKINDEX.tar.gz
(1/13) Installing fstrm (0.6.1-r1)
(2/13) Installing krb5-conf (1.0-r2)
(3/13) Installing libcom_err (1.46.6-r0)
(4/13) Installing keyutils-libs (1.6.3-r1)
(5/13) Installing libverto (0.3.2-r1)
(6/13) Installing krb5-libs (1.20.1-r0)
(7/13) Installing nghttp2-libs (1.51.0-r1)
(8/13) Installing protobuf-c (1.4.1-r1)
(9/13) Installing libuv (1.44.2-r0)
(10/13) Installing xz-libs (5.2.9-r0)
(11/13) Installing libxml2 (2.10.4-r0)
(12/13) Installing bind-libs (9.18.19-r0)
(13/13) Installing bind-tools (9.18.19-r0)
Executing busybox-1.35.0-r29.trigger
OK: 16 MiB in 28 packages
/ # ping 30x.example.com
ping: bad address '30x.example.com'
```

**Comment by Aaron-Ritter** [comment]
thanks for clarifying!

**Comment by nextgens** [comment]
Just a few closing words on this: To hit the bug and get the bounce, you need two conditions to be met:

- have a DNS zone that is **not DNSSEC signed**
- have replies for both A and AAAA records of any MX that are either NODATA or too big to fit in a single DNS packet

In real life this is very rare... and why we only have seen it reported for Microsoft which seems to be of the opinion that sending 50+ A responses occasionally (when they feel like it), from a non DNSSEC signed zone and no AAAA record is a sane thing to do.

Fix-wise, we do not consider that backporting a newer musl to 2.0 to address this bug (which occurs very rarely and does tell the user that their email has bounced) is the right thing to do. Fixing it requires updating the alpine base image which will pull a lot of newer packages in what is effectively our stable release...

That said, we are considering doing exactly that for other reasons (clamav being broken). If #2963 ends up being merged it will be fixed in 2.0.
